.github/actions/setup-microceph/action.yml

```diff
@@ -51,17 +55,40 @@ runs:
           }
           trap cleanup ERR HUP INT TERM
 
+          ephemeral_disk="${{ steps.free_ephemeral_disk.outputs.ephemeral_disk }}"
           sudo snap install microceph --channel "${{ inputs.microceph-channel }}"
           sudo microceph cluster bootstrap
-          sudo microceph.ceph config set global osd_pool_default_size 1
+          sudo microceph.ceph config set global mon_allow_pool_size_one true
           sudo microceph.ceph config set global mon_allow_pool_delete true
+          sudo microceph.ceph config set global osd_pool_default_size 1
           sudo microceph.ceph config set global osd_memory_target 939524096
           sudo microceph.ceph osd crush rule rm replicated_rule
           sudo microceph.ceph osd crush rule create-replicated replicated default osd
           for flag in nosnaptrim nobackfill norebalance norecover noscrub nodeep-scrub; do
               sudo microceph.ceph osd set $flag
           done
-          sudo microceph disk add --wipe "${{ steps.free_ephemeral_disk.outputs.ephemeral_disk }}"
+
+          # If there is more than one OSD, set up partitions.
+          if [ ${{ inputs.osd-count }} -gt 1 ]; then
+            sudo parted "${ephemeral_disk}" --script mklabel gpt
+
+            for i in $(seq 1 "${{ inputs.osd-count }}"); do
+              # Create equal sized partitions for each OSD.
+              min="$(( (${i}-1) *  100 / ${{ inputs.osd-count }} ))"
+              max="$(( ${i} * 100 / ${{ inputs.osd-count }} ))"
+              sudo parted "${ephemeral_disk}" --script mkpart primary "${min}%" "${max}%"
```

Comment:
`man parted` doesn't specify what kind of alignment it does by default so to be sure we get the the optimal one, could you add `--align optimal` in there?